Source/SwiftLintBuiltInRules/Rules/Idiomatic/UnneededBreakInSwitchRule.swift

```diff
@@ -45,23 +45,118 @@ struct UnneededBreakInSwitchRule: SwiftSyntaxRule, ConfigurationProviderRule {
         UnneededBreakInSwitchRuleVisitor(viewMode: .sourceAccurate)
     }
+
+    func makeRewriter(file: SwiftLintCore.SwiftLintFile) -> SwiftLintCore.ViolationsSyntaxRewriter? {
+        UnneededBreakInSwitchRewriter(
+            locationConverter: file.locationConverter,
+            disabledRegions: disabledRegions(file: file)
+        )
+    }
 }
 
 private final class UnneededBreakInSwitchRuleVisitor: ViolationsSyntaxVisitor {
     override func visitPost(_ node: SwitchCaseSyntax) {
-        guard node.statements.count > 1,
-              let statement = node.statements.last,
-              let breakStatement = statement.item.as(BreakStmtSyntax.self),
-              breakStatement.label == nil else {
+        guard node.containsUnneededBreak(),
+              let statement = node.statements.last else {
             return
         }
 
         violations.append(statement.item.positionAfterSkippingLeadingTrivia)
     }
 }
+
+private final class UnneededBreakInSwitchRewriter: SyntaxRewriter, ViolationsSyntaxRewriter {
+    private(set) var correctionPositions: [SwiftSyntax.AbsolutePosition] = []
+    let locationConverter: SourceLocationConverter
+    let disabledRegions: [SourceRange]
+
+    init(locationConverter: SourceLocationConverter, disabledRegions: [SourceRange]) {
+        self.locationConverter = locationConverter
+        self.disabledRegions = disabledRegions
+    }
+
+    override func visit(_ node: SwitchCaseSyntax) -> SwitchCaseSyntax {
+        guard let statement = node.statements.last,
+              node.containsUnneededBreak(),
+              !node.isContainedIn(regions: disabledRegions, locationConverter: locationConverter) else {
+            return super.visit(node)
+        }
+
+        correctionPositions.append(statement.item.positionAfterSkippingLeadingTrivia)
+
+        let trivia = statement.item.leadingTrivia + statement.item.trailingTrivia
+
+        let stmts = node.statements.removingLast()
+        let secondLast = stmts.last!
+
+        let newNode = node
+            .with(\.statements, stmts)
+            .with(\.statements.trailingTrivia, secondLast.item.trailingTrivia + trivia)
+            .trimmed { !$0.isComment }
+            .formatted()
+            .as(SwitchCaseSyntax.self)!
+
+        return super.visit(newNode)
+    }
+}
+
+private extension SwitchCaseSyntax {
+    func containsUnneededBreak() -> Bool {
+        guard statements.count > 1,
+              let breakStatement = statements.last!.item.as(BreakStmtSyntax.self),
+              breakStatement.label == nil else {
+            return false
+        }
+
+        return true
+    }
+}
+
+private extension TriviaPiece {
+    var isComment: Bool {
+        switch self {
+        case .lineComment, .blockComment, .docLineComment, .docBlockComment:
+            return true
+        default:
+            return false
+        }
+    }
+}
+
+private let lineComment = "// line comment"
```

Comment:
What's the benefit of having these comments extracted into variables? To make the examples easier to understand I'd prefer have them inlined.

Reply:
Fixed in c30d153